### DISCUSSION TITLE
Stereochemistry inference has changed since 2022.9.3

### DISCUSSION THREAD

**Discussion by MaybeJustJames** [rdkit/rdkit#7668] [Q&A] [answered]
First, I have no chemistry training so I apologise upfront for my ignorance.

I'm updating my application which uses rdkit to render 2d molecules from Python 3.11 to 3.12. As part of this I'm moving rdkit from version 2022.9.3 to the most recent release on pypi (2024.3.3 at the time of writing).

I used to be able to produce beautiful molecule images with stereochemistry that look like this:
![2022 9 3](https://github.com/user-attachments/assets/8324f1ca-6cce-46b1-8a6f-3b28ff297f1f)

But after updating to rdkit 2024.3.3 the same code and input produces this image:
![2024 3 3](https://github.com/user-attachments/assets/4d1b1c80-d702-482e-b6d2-5c05cae658c8)

...and some  warning messages:

```
[14:33:48] Warning: ambiguous stereochemistry - zero final chiral volume - at atom 2 ignored
[14:33:48] Warning: ambiguous stereochemistry - zero final chiral volume - at atom 3 ignored
[14:33:48] Warning: ambiguous stereochemistry - zero final chiral volume - at atom 4 ignored
[14:33:48] Warning: ambiguous stereochemistry - zero final chiral volume - at atom 5 ignored
[14:33:48] Warning: ambiguous stereochemistry - zero final chiral volume - at atom 1 ignored
```

The code I use is this:

```python
def _quick_draw(filename: str) -> None:
    mol = Chem.MolFromMolFile(filename, sanitize=False, removeHs=False)
    Chem.SanitizeMol(mol, catchErrors=True)
    Chem.Kekulize(mol)
    Chem.AssignStereochemistry(mol, force=True)
    mol_conf = mol.GetConformer()
    Chem.WedgeMolBonds(mol, mol_conf)
    AllChem.Compute2DCoords(mol)

    from rdkit.Chem import Draw

    img = Draw.MolToImage(mol)
    img.save("mol.png")
```

And the input MOL file is this:

```
"alpha-D-glucopyranose"
  biocyc  0721211424

 12 12  0  0  0  0  0  0  0  0999 V2000
  148.4852   71.8515    0.0000 C   0  0  0  0  0  0  0
  358.7525  -50.7030    0.0000 C   0  0  0  0  0  0  0
  358.7525 -295.7525    0.0000 C   0  0  0  0  0  0  0
  571.6634 -419.6733    0.0000 C   0  0  0  0  0  0  0
  786.0297 -295.7228    0.0000 C   0  0  0  0  0  0  0
  786.0297  -50.6733    0.0000 C   0  0  0  0  0  0  0
  -39.6832  -86.6733    0.0000 O   0  0  0  0  0  0  0
  148.4852 -419.6733    0.0000 O   0  0  0  0  0  0  0
  571.6930 -664.7228    0.0000 O   0  0  0  0  0  0  0
  999.0000 -419.6139    0.0000 O   0  0  0  0  0  0  0
  999.0000   71.8812    0.0000 O   0  0  0  0  0  0  0
  571.6634   71.8812    0.0000 O   0  0  0  0  0  0  0
  3  8  1  6  0  0  0
  4  9  1  1  0  0  0
  5 10  1  6  0  0  0
  6 11  1  6  0  0  0
  2  1  1  1  0  0  0
  3  2  1  0  0  0  0
  4  3  1  0  0  0  0
  5  4  1  0  0  0  0
  6  5  1  0  0  0  0
 12  6  1  0  0  0  0
  2 12  1  0  0  0  0
  7  1  1  0  0  0  0
M  END
```

I have no control over the input MOL files so adjusting them is not possible. Is there a way I can recover the beautiful drawing with stereochemistry? What am I doing wrong? Is there a way I can ask rdkit to behave like it used to in 2022?

Thanks for your help!
James

**Comment by MaybeJustJames** [comment]
I suppose this from the [2023.9.1 release notes](https://github.com/rdkit/rdkit/releases/tag/Release_2023_09_1) is what changed,

> The algorithm for perception of atomic stereochemistry from 2D structures has
> been rewritten. The new algorithm is more accurate, which results in some
> differences in perceived stereo between this release and the previous ones.

Does this mean there's no way to recover the beautiful stereochemistry on this MOL file?

**Comment by greglandrum** [comment, marked as answer]
@MaybeJustJames this was due to a problem caused by the "unusual" bond lengths in your input structure. It was [fixed in v2024.03.4](https://github.com/rdkit/rdkit/issues/7509), which is available from conda-forge

**Reply by MaybeJustJames** [reply]
Is it possible to publish >=2024.03.4 to Pypi? Using conda is a significant obstacle for my application.

**Comment by MaybeJustJames** [comment]
Perfect! Thank you!!
